src/sbvr-api/hooks.ts

```diff
@@ -0,0 +1,77 @@
+import * as _ from 'lodash'
+import * as Promise from 'bluebird'
+import { settleMapSeries } from './control-flow'
+
+type RollbackAction = () => any
```

Comment:
This should be `() => void | Promise<void>` since we don't use the return value (outside of waiting on a promise), and as such typing it as void will help avoid issues where people return other things by accident or because they expect them to do something